ui/src/app/machines/views/MachineDetails/MachineNetwork/InterfaceFormTable/InterfaceFormTable.tsx

```diff
@@ -0,0 +1,158 @@
+import type { ReactNode } from "react";
+
+import { MainTable, Spinner } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+
+import DHCPColumn from "../NetworkTable/DHCPColumn";
+import FabricColumn from "../NetworkTable/FabricColumn";
+import IPColumn from "../NetworkTable/IPColumn";
+import NameColumn from "../NetworkTable/NameColumn";
+import PXEColumn from "../NetworkTable/PXEColumn";
+import SpeedColumn from "../NetworkTable/SpeedColumn";
+import SubnetColumn from "../NetworkTable/SubnetColumn";
+import TypeColumn from "../NetworkTable/TypeColumn";
+
+import TableHeader from "app/base/components/TableHeader";
+import machineSelectors from "app/store/machine/selectors";
+import type {
+  NetworkInterface,
+  NetworkLink,
+  Machine,
+} from "app/store/machine/types";
+import { getInterfaceName } from "app/store/machine/utils";
+import type { RootState } from "app/store/root/types";
+
+// TODO: This should eventually extend the react-components table row type
+// when it has been migrated to TypeScript.
+type NetworkRow = {
+  columns: { className?: string; content: ReactNode }[];
+  key: NetworkInterface["name"];
+};
+
+const generateRow = (machine: Machine, nic: NetworkInterface): NetworkRow => {
+  const name = getInterfaceName(machine, nic);
+  return {
+    columns: [
+      {
+        content: <NameColumn nic={nic} systemId={machine.system_id} />,
+      },
+      {
+        content: <PXEColumn nic={nic} systemId={machine.system_id} />,
+      },
+      {
+        content: <SpeedColumn nic={nic} systemId={machine.system_id} />,
```

Comment:
Just noticed that the speed column is slightly misaligned when there's no icon because it's missing the little placeholder div. I think it's because of the React fragment being passed to the DoubleRow's `icon` prop.

Reply:
Thanks, well spotted.